semantic-conventions/src/opentelemetry/semconv/model/semantic_convention.py

```diff
@@ -131,31 +121,12 @@ def parse(yaml_file):
             prefix = group.get("prefix", "")
             if prefix != "":
                 validate_id(prefix, group.lc.data["prefix"])
-            position = group.lc.data["id"]
-            # Fields validation based on the type
-            if convention_type == SemanticConventionType.RESOURCE:
-                if span_kind != SpanKind.EMPTY:
-                    position = group.lc.data["span_kind"]
-                    msg = "Resources cannot have span_kind: {}".format(
-                        group.get("span_kind")
-                    )
-                    raise ValidationError.from_yaml_pos(position, msg)
-            model = SemanticConvention(
-                semconv_id=group["id"].strip(),
-                type=convention_type,
-                brief=str(group["brief"]).strip(),
-                note=group.get("note", "").strip(),
-                prefix=prefix.strip(),
-                extends=group.get("extends", "").strip(),
-                span_kind=span_kind,
-                attrs_by_name=SemanticAttribute.parse(prefix, group.get("attributes"))
-                if "attributes" in group
-                else {},
-                constraints=SemanticConvention.parse_constraint(
-                    group.get("constraints", ())
-                ),
-                _position=position,
-            )
+
+            # First, validate that the correct fields are available in the yaml
+            convention_type.validate_keys(group)
+            model = convention_type(group)
+            # Also, validate that the value of the fields is acceptable
+            model.validate_values()
```

Comment:
These lines appear to have gotten much simpler, but the complexity is moved from here to the constructor for the class that's returned from the `parse_semantic_convention_type` function. 

This is because the required parameters for each type of convention is different, and passing a single yaml node to the constructor allows the subclasses to own their unique implementations.